### PR TITLE
editoast: add utoipa annotations for /train_schedule

### PR DESCRIPTION
**Pull request by laggron42**
Part of https://github.com/osrd-project/osrd/issues/3625, https://github.com/osrd-project/osrd/issues/4580

# Note

**@_front-maintainers**, I've had to remove the `timetable` tag of some endpoints because utoipa's macro doesn't let me specify multiple tags. Please let me know if that's an issue.

### CODE REVIEW

File: editoast/src/views/train_schedule/mod.rs [hunk 1]
```diff
@@ -99,13 +131,23 @@ async fn delete(db_pool: Data<DbPool>, train_schedule_id: Path<i64>) -> Result<H
...
+#[derive(Debug, Deserialize, ToSchema)]
+struct BatchDeletionRequest {
     ids: Vec<i64>,
 }
-//Delete multiple train schedule
+
+/// Delete multiple train schedules at once
+#[utoipa::path(
+    tag = "train_schedule",
+    // tag = "timetable",
```
Comment (Tristramg) [inline editoast/src/views/train_schedule/mod.rs]: If we leave it as a comment, maybe be more explicit that it should also be `timetable` but that utoipa doesn’t allow it

File: editoast/src/models/train_schedule.rs [hunk 2]
```diff
@@ -42,30 +57,40 @@ use super::{
...
     pub initial_speed: f64,
-    #[derivative(Default(value = "DieselJson(Default::default())"))]
+    #[derivative(Default)]
+    #[schema(value_type = Vec<Allowance>)]
     pub allowances: DieselJson<Vec<Allowance>>,
     #[derivative(Default(
         value = "crate::schema::rolling_stock::RollingStockComfortType::default().to_string()"
     ))]
+    #[schema(value_type = RollingStockComfortType)]
     pub comfort: String,
```
Comment (Tristramg) [inline editoast/src/models/train_schedule.rs:74]: Can’t we have RollingStockComfortType as the type for comfort ?

**Review by Tristramg — APPROVED** [review]
Looks good to me. Just one question to better understand and a vague comment about a comment

File: editoast/src/models/train_schedule.rs (hunk 2 above)
Comment (leovalais) [reply editoast/src/models/train_schedule.rs:74]: Sadly no, Diesel doesn't support enums. There are packages that seem to fill the gap but we didn't like any of those. So we planned to add enum support in ModelV2, but I've just found that crate that seems really nice: https://github.com/adwhit/diesel-derive-enum

**Review by Tguisnet — COMMENTED** [review]
LGTM for the editoast part, while you're ok with the front for tags.

**Review by clarani — APPROVED** [review]
LGTM, tested :white_check_mark:

**Review by Tguisnet — APPROVED** [review]

**Pull request merged**